crates/rust/src/bindgen.rs

```diff
@@ -877,10 +879,8 @@ impl Bindgen for FunctionBindgen<'_, '_> {
 
             Instruction::CallInterface { func, .. } => {
                 if self.async_ {
-                    let tmp = self.tmp();
-                    let result = format!("result{tmp}");
-                    self.push_str(&format!("let {result} = "));
-                    results.push(result);
+                    self.push_str(&format!("let result = "));
```

Comment:
This change wasn't necessary, but I figured I'd simplify it a bit since there's no chance of a name conflict (i.e. we don't need to use `self.tmp()` here).